core/src/test/java/com/radixdlt/rev2/nofaucet/CallPreviewWithoutFaucetTest.java

```diff
@@ -0,0 +1,185 @@
+ * of any economic or other opportunity of whatsoever nature or howsoever arising), arising
+ * out of or in connection with (without limitation of any use, misuse, of any ledger system
+ * or use made or its functionality or any performance or operation of any code or protocol
+ * caused by bugs or programming or logic errors or otherwise);
+ *
+ * A. any offer, purchase, holding, use, sale, exchange or transmission of any
+ * cryptographic keys, tokens or assets created, exchanged, stored or arising from any
+ * interaction with the Work;
+ *
+ * B. any failure in a transmission or loss of any token or assets keys or other digital
+ * artefacts due to errors in transmission;
+ *
+ * C. bugs, hacks, logic errors or faults in the Work or any communication;
+ *
+ * D. system software or apparatus including but not limited to losses caused by errors
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.rev2.nofaucet;
+
+import static com.radixdlt.environment.deterministic.network.MessageSelector.firstSelector;
+import static com.radixdlt.lang.Tuple.tuple;
+import static org.junit.Assert.assertEquals;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Module;
+import com.radixdlt.addressing.Addressing;
+import com.radixdlt.api.core.generated.api.TransactionApi;
+import com.radixdlt.api.core.generated.client.ApiException;
+import com.radixdlt.api.core.generated.models.*;
+import com.radixdlt.crypto.ECKeyPair;
+import com.radixdlt.environment.deterministic.network.MessageMutator;
+import com.radixdlt.genesis.*;
+import com.radixdlt.harness.deterministic.DeterministicTest;
+import com.radixdlt.harness.deterministic.PhysicalNodeConfig;
+import com.radixdlt.identifiers.Address;
+import com.radixdlt.modules.FunctionalRadixNodeModule;
+import com.radixdlt.modules.StateComputerConfig;
+import com.radixdlt.networks.Network;
+import com.radixdlt.rev2.Decimal;
+import com.radixdlt.rev2.ScryptoConstants;
+import com.radixdlt.rev2.protocol.ProtocolUpdateTestUtils;
+import com.radixdlt.utils.UInt32;
+import com.radixdlt.utils.UInt64;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public final class CallPreviewWithoutFaucetTest {
```

Comment:
Out of interest, why didn't we use a test under `/api/core`, e.g. creating a `TransactionCallPreviewTest.java` similiar to `TransactionPreviewTest.java`?

In fact, can I request that we re-write the test there?

Reply:
You can then configure the test to run different scenarios, as per `test_core_api_can_submit_and_commit_transaction_after_running_all_scenarios`

Reply:
I'll move test there, but this is rather organizational change. This test is specific in the sense that it has different genesis, hence reusing existing classes gets quite inconvenient.

Reply:
> I'll move test there, but this is rather organizational change. This test is specific in the sense that it has different genesis, hence reusing existing classes gets quite inconvenient.

`test_core_api_can_submit_and_commit_transaction_after_running_all_scenarios` which I mentioned above uses a different genesis alongside various helpers and builders which reduces the boilerplate for such changes. It can be used as an example for such a change.